hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/CalciteSqlOptimizer.java

```diff
@@ -710,6 +712,12 @@ public static PhysicalRel uniquifyScans(PhysicalRel rel) {
         AssignDiscriminatorToScansRule rule = new AssignDiscriminatorToScansRule();
         hepProgramBuilder.addRuleInstance(rule);
 
+        // Also approach Calc-Limit transposition rule,
+        // since we support only top-level LIMIT...OFFSET,
+        // there might be the only occurence to use that rule.
+        // Also, this rule has perfect use-case exactly for heuristic planner.
+        hepProgramBuilder.addRuleInstance(CalcLimitTransposeRule.INSTANCE);
```

Comment:
Such a change demands changing the method name, it's no longer just "uniquifying of scans". E.g. `postPhysicalOptTransformations`.

```suggestion
        // The Calc-Limit-Transpose rule pushes the LimitRel up before a CalcRel.
        // We rely on this in CreateDagVisitor when handling LimitRel - it must be a
        // direct input of the RootRel, there cannot be a Calc in between.
        hepProgramBuilder.addRuleInstance(CalcLimitTransposeRule.INSTANCE);
```